gdal/frmts/vrt/pixelfunctions.cpp

```diff
@@ -873,6 +873,40 @@ static CPLErr dB2PowPixelFunc( void **papoSources, int nSources, void *pData,
                               nPixelSpace, nLineSpace, 10.0, 10.0);
 }  // dB2PowPixelFunc
 
+static CPLErr PowPixelFunc( void **papoSources, int nSources, void *pData,
+                               int nXSize, int nYSize,
+                               GDALDataType eSrcType, GDALDataType eBufType,
+                               int nPixelSpace, int nLineSpace, char **papszArgs ) {
+    /* ---- Init ---- */
+    if( nSources != 1 ) return CE_Failure;
+    if( GDALDataTypeIsComplex( eSrcType ) ) return CE_Failure;
+
+    const char *pszPower = CSLFetchNameValue(papszArgs, "power");
+    if ( pszPower == nullptr ) return CE_Failure;
```

Comment:
an explicit CPLError(CE_Failure, CPLE_AppDefined, ...) error message would be appropriate